pkg/loadbalancer/experimental/backend.go

```diff
@@ -109,47 +149,110 @@ func (be *Backend) TableRow() []string {
 	return []string{
 		be.StringWithProtocol(),
 		state,
-		showInstances(be.Instances),
+		showInstances(be),
+		showShadows(be),
 		be.NodeName,
 		strconv.FormatUint(uint64(be.ZoneID), 10),
 	}
 }
 
-func showInstances(instances part.Map[loadbalancer.ServiceName, BackendInstance]) string {
+func showInstances(be *Backend) string {
 	var b strings.Builder
-	count := instances.Len()
-	for name, inst := range instances.All() {
-		b.WriteString(name.String())
+	for k, inst := range be.PreferredInstances() {
+		b.WriteString(k.ServiceName.String())
 		if inst.PortName != "" {
 			b.WriteString(" (")
 			b.WriteString(string(inst.PortName))
 			b.WriteRune(')')
 		}
-		count--
-		if count > 0 {
-			b.WriteString(", ")
+		b.WriteString(", ")
+	}
+	full := b.String()
+	return full[:len(full)-2] // drop trailing ", "
```

Comment:
Well spotted! Can't revert because we can't initialize `count := instances.Len()` when using `be.PreferredInstances()`. I'll use `strings.TrimSuffix`.